recipes/influxdb-cxx/all/conanfile.py

```diff
@@ -0,0 +1,101 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import get, copy, rmdir
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+import os
+
+required_conan_version = ">=1.53.0"
+
+class InfluxdbCxxConan(ConanFile):
+    name = "influxdb-cxx"
+    description = "InfluxDB C++ client library."
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/offa/influxdb-cxx"
+    topics = ("influxdb", "influxdb-client")
+    settings = "os", "arch", "compiler", "build_type"
+    package_type = "library"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "boost": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "boost": True,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "8",
+            "clang": "7",
+            "apple-clang": "12",
+            "Visual Studio": "16",
+            "msvc": "192",
+        }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("cpr/1.10.4")
+        if self.options.boost:
+            self.requires("boost/1.82.0")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        # BUILD_SHARED_LIBS is defined explicitly in CMakeLists.txt
+        tc.cache_variables["BUILD_SHARED_LIBS"] = self.options.shared
+        tc.cache_variables["INFLUXCXX_TESTING"] = False
+        tc.cache_variables["INFLUXCXX_WITH_BOOST"] = self.options.boost
+        if self.options.shared:
+            tc.preprocessor_definitions["InfluxDB_EXPORTS"] = 1
```

Comment:
This one looks a little unusual - is it not possible to build the shared variant without the consumers defining preprocessor definitions externally?

Reply:
This shouldn't be necessary as the define is handled by CMakes GenerateExportHeader. However, there are reports of issues on MSVC. I wasn't able to reproduce the issue, but I'm no MSVC dev.

Reply:
Thanks @toge  - there is indeed a pre-existing issue, I've updated the recipe to reference it

I've had a chance to investigate the root cause and made a comment here:
https://github.com/offa/influxdb-cxx/issues/194#issuecomment-1935661442